src/Game.cpp

```diff
@@ -31,6 +31,164 @@
 #include "UnkObjects/unk6C2E88.hpp"
 #include "UnkObjects/unk6C3250.hpp"
 #include "UnkObjects/unk6C3890.hpp"
+#include "UnkObjects/unk8600F8.hpp"
+#include "config.hpp"
+
+static void sub_612A80(char *args) {
+    glo_8600F0 = 0;
+    glo_8600EC = nullptr;
+    char *esi = args;
+    if (*esi != '\0') {
+        esi = std::strchr(args, ' ');
+        if (esi == nullptr) {
+            glo_8600F8.loc_611DC0();
+            return;
+        }
+    }
+
+    if (*esi == '\0') {
+        glo_8600F8.loc_611DC0();
+        return;
+    }
+
+    *esi = '\0';
+    glo_8600EC = (char**)MC2_MALLOC(4);
+
+    uint32_t length = strlen(args);
+    *glo_8600EC = MC2_STRDUP(args);
+
+    *esi = ' ';
+    esi++;
+
+    if (*esi == '\0') {
+        glo_8600F8.loc_611DC0();
+        return;
+    }
+
+    for (char *cur = esi; *cur != '\0'; ++cur) {
+        if (*cur != '-')
+            continue;
+
+        glo_8600F0++;
+
+        char *nextSeperator = cur;
+        while (*nextSeperator != '\0') {
+            if (*nextSeperator == ' ')
+                break;
+            if (*nextSeperator == '=')
+                break;
+            nextSeperator++;
+        }
+        char *key = cur + 1;
+        char *value_str = cur;
+        if (*nextSeperator != '\0') {
+            *nextSeperator = '\0';
+            value_str = nextSeperator + 1;
+        }
+        unk_612150 *value = glo_8600F8.sub_6124A0(key);
+
+        if (value != nullptr) {
+            cur = value_str;
+            continue;
+        }
+
+        value = (unk_612150*)MC2_MALLOC(sizeof(unk_612150));
```

Comment:
While I'm ok being weaker on this rule for things like char arrays (such as the one above), I'm going to be more strict about it here: You should really be using the new operator with the placement parameter (see [this](http://en.cppreference.com/w/cpp/language/new) and [this](http://en.cppreference.com/w/cpp/memory/new/operator_new) for documentation, and [this](https://github.com/LRFLEW/OpenMC2/blob/e7b009cda089ca73620396b6cb0cea889bbc6b5b/src/Game.cpp#L214) for an example). The reason is that using this guarantees that the constructor gets called (if we implement one). The plan is that now that `MC2_FREE` has been identified, I want to hook `MC2_MALLOC` and `MC2_FREE` to call `operator new` and `operator delete` in the OpenRCT2 context, so that we can just start using `new` and `delete` directly.